### ISSUE TITLE
how get the symbol file when build command is 'ohrs build --arch aarch --release -- -Z build-std=std -Z build-std-features="optimize_for_size"'

### ISSUE THREAD

**Issue by SaigyoujiNexas** [ohos-rs/example#26]
I use command below to build my project as a dylib :
`
ohrs build --arch aarch --release -- -Z build-std=std -Z build-std-features="optimize_for_size"
`
and Cargo.toml below:
`
[package]
name = "im_rust_ts"
version.workspace = true
edition.workspace = true

[lib]
crate-type = ["cdylib"]

[dependencies]
lazy_static = { workspace = true }
log = { workspace = true }
once_cell = { workspace = true }
serde_json = { workspace = true }
serde = { workspace = true }
http = { workspace = true }
base64 = { workspace = true }
dyn-clone = { workspace = true }
# Default enable napi4 feature, see https://nodejs.org/api/n-api.html#node-api-version-matrix
napi = { workspace = true, optional = true }
napi-derive = { workspace = true, optional = true }
napi-ohos = { workspace = true, optional = true }
napi-derive-ohos = { workspace = true, optional = true }

[build-dependencies]
napi-build = { workspace = true, optional = true }
napi-build-ohos = { workspace = true, optional = true }

[profile.release]
lto = true
strip = true 

`
When I use the output .so file as other application component, the crash occured. I try to debug it, but the lack of symbol make me debug so hard because I can not locate which line crashed by use the objdump.

**Comment by SaigyoujiNexas** [comment]
> please build project without `--release`

but I need provide my package for other team, I can not build a debug package, because it make package size huge

**Comment by richerfu** [comment]
These are some common solutions.
You need to set `debug` with `Cargo.toml`

```toml
[profile.release]
lto = true
strip = true
debug = true
```

And save your debug info with  `objcopy`:
```bash
$OHOS_NDK_HOME/native/llvm/bin/llvm-objcopy --only-keep-debug ./dist/arm64-v8a/libhello.so ./hello.debug
```

In fact, this is a general solution for rust, which has nothing to do with ohos-rs itself.

**Comment by SaigyoujiNexas** [comment]
> These are some common solutions. You need to set `debug` with `Cargo.toml`
> 
> ```toml
> [profile.release]
> lto = true
> strip = true
> debug = true
> ```
> 
> And save your debug info with `objcopy`:
> 
> ```shell
> $OHOS_NDK_HOME/native/llvm/bin/llvm-objcopy --only-keep-debug ./dist/arm64-v8a/libhello.so ./hello.debug
> ```
> 
> In fact, this is a general solution for rust, which has nothing to do with ohos-rs itself.

no, when I set strip = false, debug = true, use the above command to compile, the objdump file is looks like that.
<img width="1436" alt="image" src="https://github.com/user-attachments/assets/e604e161-836c-4da6-a4cd-f82c9d3f7a5b">


when I delete "--release", the package size become to 150MB, but the objdump file have all message I need.
<img width="1188" alt="image" src="https://github.com/user-attachments/assets/d667b0fc-bff7-4f08-9e04-c5ce49f86d6d">

**Comment by SaigyoujiNexas** [comment]
> You need to save your debug info with `debug=true` and use debug file to locate the source code. It's impossible for us to get the smallest binary size with source info.您需要使用`debug=true`保存调试信息并使用调试文件来定位源代码。我们不可能通过源信息获得最小的二进制大小。

I got. it could not have all debug info such as line number, and what rust code relative this asm code, which I just can relay on remaining debug info( struct name, method reference expression invoke,  method signature etc. ) when compile by
```
ohrs build --release
```
right?

**Comment by richerfu** [comment]
Yep
